packages/powersync/lib/src/bucket_storage.dart

```diff
@@ -300,7 +303,7 @@ class BucketStorage {
       // |--SEARCH r USING INDEX ps_oplog_by_row (row_type=? AND row_id=?)
       // `--USE TEMP B-TREE FOR GROUP BY
       // language=DbSqlite
-      var stmt = db.prepare(
+      var opRows = await tx.execute(
```

Comment:
This change can have a big impact on memory usage. The previous version did it in batches - never keeping more than 10k rows in memory at a time (which could actually be reduced to a much smaller number / replaced saving individual rows at a time). This version keeps the entire resultset in memory, which could be massive for some use cases (100k+ rows).

Is it feasible to have separate implementations for web vs native as a stop-gap here, until we move to using the sqlite extension?

Reply:
Yes, it is feasible. I've now implemented a `NativeBucketStorage` class which extends `BucketStorage`. This uses the sqlite connection directly to apply bucket changes.